src/main/java/com/ericsson/gerrit/plugins/eiffel/git/CommitInformation.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * <strong>Note:</strong> This class uses injects. To use this class the caller must inject it,
+ * otherwise the injection framework cannot instantiate this class properly.
+ *
+ */
+public class CommitInformation {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CommitInformation.class);
+    private CommitsCollection commitsCollection;
+    private ProjectsCollection projectsCollection;
+
+    /**
+     * Package private constructor with injection for testing reasons. Injecting directly as field
+     * variables makes it harder to test
+     */
+    @Inject
+    CommitInformation(CommitsCollection commitsCollection, ProjectsCollection projectsCollection) {
+        this.commitsCollection = commitsCollection;
+        this.projectsCollection = projectsCollection;
+    }
+
+    /**
+     * Fetches a list of parents SHA from a commit. An empty list will be returned if any error
+     * occurs.
+     *
+     * @param commitId    The SHA of the commit to operate on
+     * @param projectName The name of the project which the commits resides in
+     * @return A list of parents SHAs
+     */
+    public List<String> getParentsSHAs(final String commitId, final String projectName) {
+        List<RevCommit> parents = getParents(commitId, projectName);
+        List<String> parentsSHAs = getSHAs(parents);
+        return parentsSHAs;
+    }
+
+    /**
+     * Will return a list of parents for a given commit. If the collections cannot find the project
+     * or commit, an empty list will be returned.
+     */
+    private List<RevCommit> getParents(final String commitId, final String projectName) {
+        List<RevCommit> parents = Collections.emptyList();;
+
+        try {
+            parents = getParentsFromCommit(commitId, projectName);
+        } catch (UnprocessableEntityException e) {
+            LOGGER.error(String.format("Cannot find or load the project %s", projectName), e);
+        } catch (ResourceNotFoundException e) {
+            LOGGER.error(String.format("Cannot find the commit %s", commitId), e);
+        } catch (IOException e) {
+            LOGGER.error(
+                    String.format("Error finding the commit for %s in %s", commitId, projectName),
+                    e);
+        }
+        return parents;
+    }
+
+    private List<RevCommit> getParentsFromCommit(final String commitId, final String projectName)
+            throws UnprocessableEntityException, IOException, ResourceNotFoundException {
+
+        ProjectResource projectResource = projectsCollection.parse(projectName, true);
+        CommitResource commitResource = commitsCollection.parse(projectResource,
+                IdString.fromDecoded(commitId));
+        RevCommit commit = commitResource.getCommit();
+        RevCommit[] parents = commit.getParents();
+
+        return Arrays.asList(parents);
+    }
+
+    private List<String> getSHAs(List<RevCommit> parents) {
+        List<String> parentsSha = parents.stream()
```

Comment:
Since streams is usually slower and harder to read then normal for loops, could you rewrite this to a normal for loop.

Reply:
Could then be written as:
```

        ArrayList<String> arrayList = new ArrayList();
        parents.forEach(recCommit -> arrayList.add(recCommit.getName()));
```
```

        ArrayList<String> arrayList = new ArrayList();
        for (RevCommit revCommit : parents) {
            arrayList.add(revCommit.getName());
        }
```

Which one of the three do you prefer?

Reply:
After asking around.
`        ArrayList<String> arrayList = new ArrayList();
        for (RevCommit revCommit : parents) {
            arrayList.add(revCommit.getName());
        }`
Seems to be what most people think is easier to read.

Reply:
Keep or change?

Reply:
If you can find any other place in the project where streams is used please keep it, if all other loops is for loops, then I would say keep it the same within project.

Reply:
Just the introduction of streams doesn't mean you can't write for loops. I'll keep it